### PR TITLE
apply pyupgrade --py38-plus --keep-runtime-typing

### PR DESCRIPTION
**Pull request by altendky**
### Purpose:

1) Bring more consistency to our code.
2) Automatically update our code base as we drop old Python versions occasionally such as in https://github.com/Chia-Network/chia-blockchain/pull/14611.

### Current Behavior:

pyupgrade has not been applied at all.

### New Behavior:

pyupgrade is applied as part of pre-commit.

### Testing Notes:

General consideration of bulk changes.

### Draft for:
- [x] discussion

### CODE REVIEW

**Comment by arvidn** [comment]
I think this would be reasonable. Are we confident the changes of exception types are correct and won't change behavior?

**Comment by altendky** [comment]
I am self-reviewing and marking changed files I've looked over so that GitHub can show me what has changed in each catch up I do.  But yes, someone else reviewing in detail when we actually want to merge this would be good.

**Comment by emlowe** [comment]
This looks good to me - I think a secondary review will be worthwhile.

**Review by arvidn — APPROVED** [review]

File: chia/daemon/server.py [hunk 1]
```diff
@@ -1079,7 +1079,7 @@ async def _start_plotting(self, id: str, loop: asyncio.AbstractEventLoop, queue:
 
             self._post_process_plotting_job(config)
 
-        except (subprocess.SubprocessError, IOError):
+        except (subprocess.SubprocessError, OSError):
```
[the remark this answers is not in the document]
Comment (altendky) [reply chia/daemon/server.py:1082]: https://docs.python.org/3/library/exceptions.html#OSError
> Changed in version 3.3: EnvironmentError, IOError, WindowsError, socket.error, select.error and mmap.error have been merged into OSError, and the constructor may return a subclass.

not sure exactly how to read that.  says merged but then says may return a subclass

they don't actually list IOError in the exception hierarchy https://docs.python.org/3/library/exceptions.html#exception-hierarchy

**Pull request merged**